content/en/04-tutorials/01-pixie-101/07-kafka-monitoring.md

```diff
@@ -0,0 +1,182 @@
+
+Let's use Pixie to see a graph of producers and consumers for each Kafka topic.
+
+1. Select `px/kafka_overview` from the script drop-down menu.
+
+2. Select the drop-down arrow next to the `namespace` argument, type `px-kafka`, and press `Enter` to re-run the script.
+
+::: div image-xl relative
+<svg title='' src='use-case-tutorials/kafka/kafka-overview.png'/>
+:::
+
+> Pixie is able to automatically discover and list all clients and topics in the cluster.
+
+> This graph shows us that our microservice app has 1 topic, named `order`, with 1 producer and 2 consumers.
+
+3. Hover over an edge on the graph to see throughput and total record Bytes for a producer or consumer. The thickness of the edge indicates an increase in throughput.
+
+> By examining the edges of the graph, we can see that the producer is producing to the order topic at about `450 B/s`. The `shipping` client is consuming at the same rate of `450 B/s`. The `invoicing` client is only consuming at `200 B/s` and is falling behind. Since this is a demo application, the traffic throughput is lower than what you'd typically see in a production application.
+
+> The **Kafka Topics** table below the graph summarizes the same high-level topic information, and also includes the number of topic partitions.
+
+4. Scroll down the page to see tables listing the brokers, producer and consumer pods.
+
+::: div image-xl relative
+<svg title='' src='use-case-tutorials/kafka/kafka-overview-2.png'/>
+:::
+
+> The **Kafka Producer Pods** and **Kafka Consumer Pods** tables confirm that the `invoicing` client is consuming messages at a slower speed than they are being produced.
+
+> The `shipping` client has sent 292 `Fetch` requests while the `invoicing` client has sent much fewer (only 192 `Fetch` requests). This again indicates that something is wrong.
+
+> When debugging Kafka problems, you may want to check the health of the clients involved. Pixie makes it easy to switch between higher-level Kafka system metrics and the lower-level client infra metrics.
+
+5. To check the health of the `invoicing` client, click on the `px-kafka/invoicing-*` pod name in the **POD** column of the **Kafka Consumer Pods** table.
+
+> Clicking on any pod name will take you to the `px/pod` script for that pod. This script shows an overview of the pod's CPU usage, network traffic and throughput, disk, and memory usage.
+
+> In this case, CPU utilization is low, which means resource starvation is not causing the `invoicing` client to consume messages at a slower rate.
+
+## Kafka Events with Metadata
+
+Let's inspect the raw Kafka requests flowing through the cluster.
+
+1. Select `px/kafka_data` from the script drop-down menu.
+
+> Pixie is able to [automatically trace](https://docs.px.dev/about-pixie/pixie-ebpf/) all messages flowing through your cluster, identify the ones using the Kafka protocol and parse the message metadata. This script shows a sample of the most recent Kafka events in the cluster.
```

Comment:
nit: Pixie automatically traces all messages flowing through your cluster, identifies the..., and parses the message...